akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventHandler.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.persistence.typed.javadsl
+
+import java.util.function.BiFunction
+
+import akka.util.OptionVal
+
+/**
+ * SAM for reacting on events having been persisted
+ *
+ * Used with [[EventHandlerBuilder]] to setup the behavior of a [[PersistentBehavior]]
+ */
+@FunctionalInterface
+trait EventHandler[Event, State] {
+  def apply(state: State, event: Event): State
+}
+
+object EventHandlerBuilder {
+  def builder[Event, State >: Null](rootEventClass: Class[Event]): EventHandlerBuilder[Event, State] =
+    new EventHandlerBuilder[Event, State](rootEventClass)
+}
+
+final class EventHandlerBuilder[Event, State >: Null](rootEventClass: Class[Event]) {
+
+  private final case class EventHandlerCase(predicate: Event ⇒ Boolean, handler: BiFunction[State, Event, State])
+
+  private var cases: List[EventHandlerCase] = Nil
+
+  private def addCase(predicate: Event ⇒ Boolean, handler: BiFunction[State, Event, State]): Unit = {
+    cases = EventHandlerCase(predicate, handler) :: cases
+  }
+
+  /**
+   * Match any event which is an instance of `E` or a subtype of `E`
+   *
+   * Throws `java.lang.IllegalArgumentException` if `eventClass` is not a subtype of the root `Event` this builder was created with
+   */
+  def matchEvent[E <: Event](eventClass: Class[E], biFunction: BiFunction[State, E, State]): EventHandlerBuilder[Event, State] = {
```

Comment:
Another place to check bounds vs Java in the compiled code

Reply:
Looks fine, Class<? extends Event>